Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -99,32 +99,24 @@ void JitArm64::GenerateAsm()
 
   if (assembly_dispatcher)
   {
-    // set the mem_base based on MSR flags
-    LDR(IndexType::Unsigned, ARM64Reg::W28, PPC_REG, PPCSTATE_OFF(msr));
-    FixupBranch physmem = TBNZ(ARM64Reg::W28, 31 - 27);
-    MOVP2R(MEM_REG,
-           jo.fastmem_arena ? memory.GetPhysicalBase() : memory.GetPhysicalPageMappingsBase());
-    FixupBranch membaseend = B();
-    SetJumpTarget(physmem);
-    MOVP2R(MEM_REG,
-           jo.fastmem_arena ? memory.GetLogicalBase() : memory.GetLogicalPageMappingsBase());
-    SetJumpTarget(membaseend);
+    ARM64Reg msr = ARM64Reg::W27;
+    LDR(IndexType::Unsigned, msr, PPC_REG, PPCSTATE_OFF(msr));
+
+    UpdateMembase();
```

Comment:
Do we really need to do UpdateMembase each time we're in the dispatcher? If you instead update the value of MEM_REG every time the msr changes, you shouldn't need up update MEM_REG here, because MEM_REG is callee-saved and isn't used for anything else in the JIT, and so should never get overwritten.

(I'm not sure if Jit64's RMEM is callee-saved.)

Reply:
Yeah you're right, we should only need it when going through enter_code right?
I tired that but it didn't work, i'll look into why eventually but for now i need to catch a plane :).

Reply:
Right, you'll need it for enter_code, but other than that it should just be needed when msr is changed.

Reply:
> (I'm not sure if Jit64's RMEM is callee-saved.)
Jit64's RMEM is callee-saved.